src/controller/SetUpCodePairer.cpp

```diff
@@ -161,8 +161,35 @@ void SetUpCodePairer::OnDiscoveredDeviceOverBleError(void * appState, CHIP_ERROR
 #endif // CONFIG_NETWORK_LAYER_BLE
 
 #if CHIP_DEVICE_CONFIG_ENABLE_DNSSD
-void SetUpCodePairer::OnDiscoveredDevice(const Dnssd::DiscoveredNodeData & nodeData)
+
+bool SetUpCodePairer::NodeMatchesCurrentFilter(const Dnssd::DiscoveredNodeData & nodeData)
 {
+    switch (currentFilter.type)
+    {
+    case Dnssd::DiscoveryFilterType::kShort:
+        return (nodeData.longDiscriminator >> 8) == currentFilter.code;
+    case Dnssd::DiscoveryFilterType::kLong:
+        return nodeData.longDiscriminator == currentFilter.code;
+    case Dnssd::DiscoveryFilterType::kVendor:
+        return nodeData.vendorId == currentFilter.code;
+    case Dnssd::DiscoveryFilterType::kDeviceType:
+        return nodeData.deviceType == currentFilter.code;
+    case Dnssd::DiscoveryFilterType::kCommissioningMode:
+        return nodeData.commissioningMode == currentFilter.code;
+    case Dnssd::DiscoveryFilterType::kNone:
+    case Dnssd::DiscoveryFilterType::kInstanceName:
+    case Dnssd::DiscoveryFilterType::kCommissioner:
+    case Dnssd::DiscoveryFilterType::kCompressedFabricId:
```

Comment:
`Dnssd::DiscoveryFilterType::kLong` and `Dnssd::DiscoveryFilterType::kShort` are the only thing handle by the `SetUpCodePairer`. Please remove the other cases and just return `false` if the type is not `kShort`/ `kLong`

Reply:
Addressed in next commit.